listenbrainz/webserver/views/api_tools.py

```diff
@@ -55,8 +55,12 @@ def _send_listens_to_queue(listen_type, listens):
     for listen in listens:
         if listen_type == LISTEN_TYPE_PLAYING_NOW:
             try:
-                expire_time = listen["track_metadata"]["additional_info"].get("duration",
-                                    current_app.config['PLAYING_NOW_MAX_DURATION'])
+                if 'duration' in listen['track_metadata']['additional_info']:
+                    expire_time = listen['track_metadata']['additional_info']['duration']
```

Comment:
This whole block bugs me. Expire_time to me suggest that an absolute time of when something happens and duration is an interval of time. While the code looks correct, it feels awkward to read.